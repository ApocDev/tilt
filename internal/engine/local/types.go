package local

import (
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"

	"github.com/tilt-dev/tilt/pkg/apis/core/v1alpha1"
)

type Cmd = v1alpha1.Cmd
type CmdList = v1alpha1.CmdList
type CmdStatus = v1alpha1.CmdStatus
type CmdSpec = v1alpha1.CmdSpec
type CmdStateWaiting = v1alpha1.CmdStateWaiting
type CmdStateTerminated = v1alpha1.CmdStateTerminated
type CmdStateRunning = v1alpha1.CmdStateRunning
type CmdServer = v1alpha1.CmdServer
type CmdServerSpec = v1alpha1.CmdServerSpec
type CmdServerStatus = v1alpha1.CmdServerStatus
type ObjectMeta = metav1.ObjectMeta
