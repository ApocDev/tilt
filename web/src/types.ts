export enum ResourceView {
  Log,
  Preview,
  Alerts,
}

export enum TriggerMode {
  TriggerModeAuto,
  TriggerModeManual,
}

export type Build = {
  Error: {} | string | null
  StartTime: string
  Log: string
  FinishTime: string
  Edits: Array<string> | null
  IsCrashRebuild: boolean
  Warnings: Array<string> | null
}

export type TiltBuild = {
  Version: string
  Date: string
  Dev: boolean
}

// what is the status of the resource in the cluster
export enum RuntimeStatus {
  Ok = "ok",
  Pending = "pending",
  Error = "error",
}

// What is the status of the resource with respect to Tilt
export enum ResourceStatus {
  BuildQueued, // in auto, if you have changed a file but an affected build hasn't started yet. In manual after you have clicked build, before it has started building
  Building,
  Error,
  Warning,
  Deploying,
  Deployed, // defer to RuntimeStatus
}

export type Resource = {
  Name: string
  CombinedLog: string
  BuildHistory: Array<any>
  CrashLog: string
  CurrentBuild: any
  DirectoriesWatched: Array<any>
  Endpoints: Array<string>
  PodID: string
  IsTiltfile: boolean
  LastDeployTime: string
  PathsWatched: Array<string>
  PendingBuildEdits: Array<string>
  PendingBuildReason: number
  PendingBuildSince: string
  ResourceInfo: {
    PodCreationTime: string
    PodLog: string
    PodName: string
    PodRestarts: number
    PodUpdateStartTime: string
    YAML: string
    PodStatus: string
    Endpoints: Array<string>
  }
  RuntimeStatus: string
  TriggerMode: TriggerMode
  HasPendingChanges: boolean
}
