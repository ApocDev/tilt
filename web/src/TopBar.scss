@import "constants";

.TopBar {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  padding-right: $sidebar-width;
  background-color: $color-gray-darkest;
  box-shadow: inset 0px -2px 10px 0px rgba($color-black, $translucent);
  z-index: $z-topBar;
  color: $color-gray-light;

  display: flex;
  flex-direction: row;
  align-items: center;
}

// Collapse/Expand
.TopBar-spacer {
  flex-grow: 1;
  height: 0;
}
.TopBar-headerDiv-snapshotURL {
  flex: 1;
  display: flex;
  margin-left: $spacing-unit * 15;
  justify-content: flex-end;
}
.TopBar-headerDiv-snapshotURL button {
  background-color: transparent;
  border: 1px solid rgba($color-white, $translucent-ish);
  color: $color-white;
  text-transform: uppercase;
}

.AlertPane-headerDiv-snapshotURL button:hover {
  background-color: $color-gray-dark;
}
