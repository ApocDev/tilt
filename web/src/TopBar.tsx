import React, { PureComponent } from "react"
import {ResourceView, Snapshot} from "./types"
import "./TopBar.scss"
import SailInfo from "./SailInfo"
import TabNav from "./TabNav"
import {Alert} from "./alerts";

type TopBarProps = {
  previewUrl: string
  logUrl: string
  alertsUrl: string
  resourceView: ResourceView
  sailEnabled: boolean
  sailUrl: string
  numberOfAlerts: number
  state: Snapshot
    handleSendSnapshot: (snapshot: Snapshot) => void

}

class TopBar extends PureComponent<TopBarProps> {
  render() {
    return (
      <div className="TopBar">
        <TabNav
          previewUrl={this.props.previewUrl}
          logUrl={this.props.logUrl}
          alertsUrl={this.props.alertsUrl}
          resourceView={this.props.resourceView}
          numberOfAlerts={this.props.numberOfAlerts}
        />
          <div className="TopBar-headerDiv-snapshotURL">
              {renderSnapshotLinkButton(this.props.state, this.props.handleSendSnapshot)}
          </div>
        <span className="TopBar-spacer">&nbsp;</span>
          <SailInfo
          sailEnabled={this.props.sailEnabled}
          sailUrl={this.props.sailUrl}
        />

      </div>
    )
  }
}

function renderSnapshotLinkButton(snapshot:Snapshot,handleSendSnapshot: (snapshot: Snapshot) => void ){ //TODO TFT
    return <button onClick={() => handleSendSnapshot(snapshot)}>Get Link</button>

}

export default TopBar
