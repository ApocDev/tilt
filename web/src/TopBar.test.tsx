import React from "react"
import renderer from "react-test-renderer"
import { MemoryRouter } from "react-router"
import { ResourceView } from "./types"
import TopBar from "./TopBar"

const fakeHandleOpenModal = () => {}

it("shows snapshot url", () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <TopBar
          logUrl="/r/foo"
          alertsUrl="/r/foo/alerts"
          facetsUrl={null}
          resourceView={ResourceView.Alerts}
          numberOfAlerts={1}
          showSnapshotButton={true}
          handleOpenModal={fakeHandleOpenModal}
          highlight={null}
        />
      </MemoryRouter>
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})

it("doesn't render snapshot button if it's a snapshot", () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <TopBar
          logUrl="/r/foo"
          alertsUrl="/r/foo/alerts"
          facetsUrl={null}
          resourceView={ResourceView.Alerts}
          numberOfAlerts={1}
          showSnapshotButton={false}
          handleOpenModal={fakeHandleOpenModal}
          highlight={null}
        />
      </MemoryRouter>
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})
