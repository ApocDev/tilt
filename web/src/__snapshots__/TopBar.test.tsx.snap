// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`doesn't render snapshot button if it's a snapshot 1`] = `
<div
  className="TopBar"
>
  <nav
    className="TabNav"
  >
    <ul>
      <li>
        <a
          className="tabLink "
          href="/r/foo"
          onClick={[Function]}
        >
          Logs
        </a>
      </li>
      <li>
        <a
          className="tabLink tabLink--errors tabLink--is-selected"
          href="/r/foo/alerts"
          onClick={[Function]}
        >
          Alerts
          <span
            className="tabLink-alertBadge"
          >
            1
          </span>
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="TopBar-tools"
  >
    
  </div>
</div>
`;

exports[`shows snapshot url 1`] = `
<div
  className="TopBar"
>
  <nav
    className="TabNav"
  >
    <ul>
      <li>
        <a
          className="tabLink "
          href="/r/foo"
          onClick={[Function]}
        >
          Logs
        </a>
      </li>
      <li>
        <a
          className="tabLink tabLink--errors tabLink--is-selected"
          href="/r/foo/alerts"
          onClick={[Function]}
        >
          Alerts
          <span
            className="tabLink-alertBadge"
          >
            1
          </span>
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="TopBar-tools"
  >
    <button
      className="TopBar-toolsButton TopBar-createSnapshotButton "
      onClick={[Function]}
    >
      <svg
        className="TopBar-snapshotSvg"
      >
        snapshot.svg
      </svg>
      <span>
        Create a 
        <br />
        Snapshot
      </span>
    </button>
  </div>
</div>
`;
